db/db_test.go

```diff
@@ -104,11 +97,7 @@ func TestBatchRollback(t *testing.T) {
 		if kvboltDB, ok := kvStore.(*boltDB); ok {
 			err = kvboltDB.batchPutForceFail(bucket1, testK1[:], testV[:])
 			assert.NotNil(err)
-		} else if kvbadgerDB, ok := kvStore.(*boltDB); ok {
```

Comment:
No need to have this if/else logic, as we only test against boltdb impl

Reply:
Also you could give up the internal func `NewOnDiskDB`

Reply:
I'll get rid of if logic